### PR TITLE
Docs: Updates for the addon documentation

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request the create addons & writing presets documentation was updated to reflect the recent updates to Storybook, effectively consolidating the addon documentation in a single place for better maintenance.

What was done:
- Updated the writing addons documentation based on the [tutorial](https://storybook.js.org/tutorials/create-an-addon/)
- Updated the required snippets/assets to factor in the item above
- Updated the writing presets documentation
- Fixed incorrect links in the documentation

@Integrayshaun and @ndelangen can both of you take a pass at this and follow up with me on any feedback that you may have? Let me know and we'll go from there.

### CODE REVIEW

File: docs/addons/addon-types.md [hunk 1]
```diff
@@ -88,6 +88,10 @@ Use this boilerplate code while writing your own preset addon.
 
 <!-- prettier-ignore-end -->
 
-<div class="aside">
-Learn more about writing your own preset addon <a href="./writing-presets">here</a>.
-</div>
+## Learn more about the Storybook addon ecosystem
+
+- Addon types for other types of addons
```
Comment (Integrayshaun) [inline docs/addons/addon-types.md:93]: Could we simplify this to "Types of addons"?

File: docs/addons/addons-api.md [hunk 2]
```diff
@@ -374,3 +374,11 @@ Hook that allows you to retrieve or update a story's [`args`](../writing-stories
 />
 
 <!-- prettier-ignore-end -->
+
+## Learn more about the Storybook addon ecosystem
+
+- [Addon types](./addon-types.md) for other types of addons
```
Comment (Integrayshaun) [inline docs/addons/addons-api.md:380]: Same as above ☝️

File: docs/addons/integration-catalog.md [hunk 3]
```diff
@@ -101,3 +101,11 @@ Not finding the recipe that you want? If it's popular in the community, our docs
 ### Request a recipe
 
 If you'd like to request a recipe, head over to the [#maintenance channel](https://discord.com/channels/486522875931656193/490070912448724992) of our community Discord and ask.
+
+## Learn more about the Storybook addon ecosystem
+
+- [Addon types](./addon-types.md) for other types of addons
```
Comment (Integrayshaun) [inline docs/addons/integration-catalog.md:107]: Same as above ☝️

**Review by ShaunEvening — COMMENTED** [review]
Looks great, João! Just a few comments in the files.

Also, can we add pnpm as an option in the shell snippets?

**Comment by ndelangen** [comment]
@jonniebigodes Is there a easy to digest preview URL I can take a look at, making this easier to review?

**Comment by jonniebigodes** [comment]
@ndelangen at this stage no, but if you're ok with it, you can follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `chore_docs_addon_changes` and let me know of any feedback

File: docs/addons/addon-types.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/addons/addon-types.md]: Sure we can and already adjusted  with the latest commit 🙏

**Comment by jonniebigodes** [comment]
That's a fantastic suggestion @Integrayshaun already added the `pnpm` variants.

**Review by ndelangen — APPROVED** [review]

**Pull request merged**